### ISSUE TITLE
Redirect back to login page (PAM auth)

### ISSUE THREAD

**Issue by JoyMonteiro** [jupyterhub/jupyterhub#996]
Hi! Thanks for using JupyterHub.

I'm running jupyterhub 0.7.1, and I'm facing an issue similar to, but not the same as, some others have posted.

I'm using the vanilla configuration, with private port set to 10110 (following something similar posted in another issue), and public port set to 5010 (I tried values of 8010, 20202 also).

The hub is running on the same LAN as I am, and is listening on all ports of the machine. If I ssh -X into the machine, open a browser on that machine, and use jupyterhub, PAM authentication works just fine, and it drops me into my home folder on that machine.

However, if I point my browser to the IP of that machine, I get a login page, but after login, it redirects me back to the login page. I'm pasting the output from running ``jupyter troubleshoot``

[I 2017-02-22 16:09:18.844 JupyterHub spawner:781] Spawning jupyterhub-singleuser '--user="joy"' '--cookie-name="jupyter-hub-token-joy"' '--base-url="/user/joy"' '--hub-host=""' '--hub-prefix="/hub/"' '--hub-api-url="http://127.0.0.1:10110/hub/api"' '--ip="127.0.0.1"' --port=35464
[W 2017-02-22 16:09:19.243 joy loader:419] Unrecognized JSON config file version, assuming version 1
[I 2017-02-22 16:09:20.048 joy manager:21] [nb_conda_kernels] enabled, 3 kernels found
[I 2017-02-22 16:09:20.424 joy __init__:35] ✓ nbpresent HTML export ENABLED
[W 2017-02-22 16:09:20.424 joy __init__:43] ✗ nbpresent PDF export DISABLED: No module named 'nbbrowserpdf'
[I 2017-02-22 16:09:20.458 joy handlers:73] [nb_anacondacloud] enabled
[I 2017-02-22 16:09:20.461 joy handlers:250] [nb_conda] enabled
[I 2017-02-22 16:09:20.463 joy notebookapp:1128] Serving notebooks from local directory: /home/joy
[I 2017-02-22 16:09:20.463 joy notebookapp:1128] 0 active kernels 
[I 2017-02-22 16:09:20.463 joy notebookapp:1128] The Jupyter Notebook is running at: http://127.0.0.1:35464/user/joy/
[I 2017-02-22 16:09:20.463 joy notebookapp:1129] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[I 2017-02-22 16:09:20.466 joy log:47] 302 GET /user/joy (127.0.0.1) 0.67ms
[I 2017-02-22 16:09:20.468 JupyterHub base:322] User joy server took 1.787 seconds to start
**[I 2017-02-22 16:09:20.468 JupyterHub orm:188] Adding user joy to proxy /user/joy => http://127.0.0.1:35464
[I 2017-02-22 16:09:20.503 JupyterHub log:100] 302 POST /hub/login?next= (@::ffff:130.237.189.166) 1839.88ms
[I 2017-02-22 16:09:20.503 JupyterHub login:91] User logged in: joy
[I 2017-02-22 16:09:20.516 JupyterHub log:100] 302 GET /hub/ (@::ffff:130.237.189.166) 1.73ms
[I 2017-02-22 16:09:20.525 JupyterHub log:100] 302 GET /login (@::ffff:130.237.189.166) 1.36ms
[I 2017-02-22 16:09:20.572 JupyterHub log:100] 200 GET /hub/login (@::ffff:130.237.189.166) 37.96ms**

I have highlighted the last bit to show what happens. If I access the hub after ssh -X (i.e, locally), After 
"User logged in: joy"
It drops me to my home folder.

Here, the server seems to be redirecting me back to the login page. Any suggestions on what I should be doing?

TIA,
Joy

**Comment by willingc** [comment]
Hi @JoyMonteiro,

Sorry for the delay in responding. I've been traveling this week.

Would you mind upgrading to JupyterHub release 0.7.2 and see if the same issue is happening. Thanks.

**Comment by JoyMonteiro** [comment]
Hello @willingc , I just tried that, and now the singleuser server crashes before starting up. Here is the traceback:

[I 2017-02-27 09:04:31.620 JupyterHub spawner:783] Spawning jupyterhub-singleuser '--user="joy"' '--cookie-name="jupyter-hub-token-joy"' '--base-url="/user/joy"' '--hub-host=""' '--hub-prefix="/hub/"' '--hub-api-url="http://127.0.0.1:10110/hub/api"' '--ip="127.0.0.1"' --port=49818
Traceback (most recent call last):
  File "/opt/anaconda3/bin/jupyterhub-singleuser", line 6, in <module>
    main()
  File "/opt/anaconda3/lib/python3.5/site-packages/jupyterhub/singleuser.py", line 322, in main
    return SingleUserNotebookApp.launch_instance(argv)
  File "/opt/anaconda3/lib/python3.5/site-packages/jupyter_core/application.py", line 267, in launch_instance
    return super(JupyterApp, cls).launch_instance(argv=argv, **kwargs)
  File "/opt/anaconda3/lib/python3.5/site-packages/traitlets/config/application.py", line 595, in launch_instance
    app.initialize(argv)
  File "<decorator-gen-7>", line 2, in initialize
  File "/opt/anaconda3/lib/python3.5/site-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/opt/anaconda3/lib/python3.5/site-packages/notebook/notebookapp.py", line 1065, in initialize
    super(NotebookApp, self).initialize(argv)
  File "<decorator-gen-6>", line 2, in initialize
  File "/opt/anaconda3/lib/python3.5/site-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/opt/anaconda3/lib/python3.5/site-packages/jupyter_core/application.py", line 239, in initialize
    self.parse_command_line(argv)
  File "/opt/anaconda3/lib/python3.5/site-packages/notebook/notebookapp.py", line 816, in parse_command_line
    super(NotebookApp, self).parse_command_line(argv)
  File "<decorator-gen-4>", line 2, in parse_command_line
  File "/opt/anaconda3/lib/python3.5/site-packages/traitlets/config/application.py", line 74, in catch_config_error
    return method(app, *args, **kwargs)
  File "/opt/anaconda3/lib/python3.5/site-packages/traitlets/config/application.py", line 512, in parse_command_line
    self.update_config(config)
  File "/opt/anaconda3/lib/python3.5/site-packages/traitlets/config/configurable.py", line 192, in update_config
    self._load_config(config)
  File "/opt/anaconda3/lib/python3.5/site-packages/traitlets/config/configurable.py", line 169, in _load_config
    warn(msg)
  File "/opt/anaconda3/lib/python3.5/contextlib.py", line 66, in __exit__
    next(self.gen)
  File "/opt/anaconda3/lib/python3.5/site-packages/traitlets/traitlets.py", line 1085, in hold_trait_notifications
    self.notify_change(change)
  File "/opt/anaconda3/lib/python3.5/site-packages/traitlets/traitlets.py", line 1127, in notify_change
    c(change)
  File "/opt/anaconda3/lib/python3.5/site-packages/jupyterhub/singleuser.py", line 147, in _user_changed
    self.log.name = change.new
AttributeError: 'dict' object has no attribute 'new'
[W 2017-02-27 09:04:41.656 JupyterHub web:1521] 500 POST /hub/login?next= (::ffff:130.237.189.166): Spawner failed to start [status=1]

Any ideas about how to fix this?

Thanks,
Joy

**Comment by JoyMonteiro** [comment]
@minrk  It works now! I tried it with 0.7.1, since we are using it in "production". Thanks!!
